packages-tests/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser/ArrayParserTest.php

```diff
@@ -40,15 +41,15 @@ public function test(string $docContent, array $expectedArrayItemNodes): void
 
     public static function provideData(): Iterator
```

Comment:
you should not change existing tests but add new ones. that way you also proof that your change is backwards compatible

Reply:
Here it's about adapting the tests to the new behavior.
Passing a literal string to ArrayItemNode to value or key parameters is still a valid case, but it doesn't mean the same thing: parsing `foo=123` and `foo="123"` will result to `ArrayItemNode('123', 'foo')` vs `ArrayItemNode(StringNode('123'), 'foo')`

Reply:
> I think it depends if the class is part of the API or internal.

agree. I don't know whether the touched files are part of the public api or not.

> Here it's about adapting the tests to the new behavior.

you changed the construct which means you broke BC. I think this changes are part of the cause, why CI fails in this PR, because packages outside this repo rely on the previous existing signature.

see e.g. https://github.com/rectorphp/rector-symfony/blob/11347bccc8b4adf8c5a40eb6acfc49aeab0b790e/src/Rector/ClassMethod/AddRouteAnnotationRector.php#L145-L199

Reply:
Yes it breaks other repos, they must be updated to.
I see this PR as a bugfix to the Doctrine annotation parsing, to better take strings into account.
The first plan idea that come to my mind looks like this:
- polish this PR so that it get merged not taking into account depending repos
- update the depending repos to make proper use of the new parsing way
- update their minimum requirement on rector to the next release of this repo
- release needed new versions
But I may be wrong, @TomasVotruba @samsonasik any feedback?

Reply:
This PR is indeed a BC break (and so will need to be released as `0.(X+1)`). But to me, it is a necessary one. Currently, all your doctrine/annotations support is actually not supporting the doctrine/annotations syntax but a different one you invented.